thorlcr/activities/rollup/throllupslave.cpp

```diff
@@ -591,25 +591,29 @@ class CRollupGroupSlaveActivity : public CSlaveActivity, public CThorDataLink
     CATCH_NEXTROW()
     {
         ActivityTimer t(totalCycles, timeActivities);
-        if (!eoi)
+        if (eoi)
+            return NULL;
+
+        loop
         {
             CThorExpandingRowArray rows(*this, queryRowInterfaces(input));
```

Comment:
I suspect this is inefficient recreating this row array each time - but I haven't checked.